src/python/bin/aqueduct

```diff
@@ -128,7 +131,24 @@ def update_server_version():
 
     execute_command([os.path.join(server_directory, "bin", "migrator"), "--type", "sqlite", "goto", "8"])
 
-def server(expose):
+def generate_welcome_message(expose_ip):
+    if not expose_ip:
+        expose_ip = "localhost"
+    apikey = get_apikey()
+    message = f"""
```

Comment:
Nit, let's make this a constant at the top of the file.

Reply:
Maybe a silly question, but this message has a few variables in it (`expose_ip`, `apikey`). How to make them a constant?

Reply:
You can put in string interpolation placeholders and then use `%` to perform the interpolation at runtime.